frontend/src/app/main/ui/workspace/shapes/text.cljs

```diff
@@ -272,49 +312,103 @@
 
         on-mount
         (fn []
-          (let [lkey1 (events/listen js/document EventType.CLICK on-click)
-                lkey2 (events/listen js/document EventType.KEYUP on-key-up)]
-            (st/emit! (dwt/assign-editor id editor)
-                      dwc/start-undo-transaction)
-
-            #(do
-               (st/emit! (dwt/assign-editor id nil)
-                         dwc/commit-undo-transaction)
-               (events/unlistenByKey lkey1)
-               (events/unlistenByKey lkey2))))
+          (when (not read-only?)
+            (let [lkey1 (events/listen js/document EventType.CLICK on-click-outside)
+                  lkey2 (events/listen js/document EventType.KEYUP on-key-up)]
+              (st/emit! (dwt/assign-editor id editor)
+                        dwc/start-undo-transaction)
+
+              #(do
+                 (st/emit! (dwt/assign-editor id nil)
+                           dwc/commit-undo-transaction)
+                 (events/unlistenByKey lkey1)
+                 (events/unlistenByKey lkey2)))))
 
         on-focus
         (fn [event]
-          (dwt/editor-select-all! editor))
+          (when (not read-only?)
+            (dwt/editor-select-all! editor)))
 
         on-change
         (mf/use-callback
          (fn [val]
-           (let [content (js->clj val :keywordize-keys true)
-                 content (first content)]
-             (st/emit! (dw/update-shape id {:content content}))
-             (reset! state val))))]
+           (when (not read-only?)
+             (let [content (js->clj val :keywordize-keys true)
+                   content (first content)]
+               (st/emit! (dw/update-shape id {:content content}))
+               (reset! state val)
+               (reset! content-var content)))))
+
+        loaded-fonts (mf/use-var 0)
+        on-load-font #(swap! loaded-fonts inc)]
 
     (mf/use-effect on-mount)
 
-    [:foreignObject {:transform (geom/transform-matrix shape)
-                     :x x :y y :width width :height height :ref self-ref}
+    (mf/use-effect
+     (mf/deps content)
+     (fn []
+       (reset! state (parse-content content))
+       (reset! content-var content)))
+
+    ;; Checks the size of the wrapper to update if it were necesary
+    (mf/use-effect
+     (mf/deps shape @loaded-fonts)
+     (fn []
+       (timers/schedule
+        250 ;; We need to wait to the text to be rendered. Is there a better alternative?
+        #(let [self-node (mf/ref-val self-ref)
+               paragraph-node (when self-node (dom/query self-node ".paragraph-set"))]
```

Comment:
@niwinz Can you think in a better alternative? I need the font to be rendered before I can check its width/height